File: bindings/python/examples/07_get_message_data.py
import iota_client
client = iota_client.Client()

message = client.get_message_data("e2daa4c6b012b615becd6c12189b2c9e701ba0d53b31a15425b21af5105fc086")
message_meta = client.get_message_metadata("e2daa4c6b012b615becd6c12189b2c9e701ba0d53b31a15425b21af5105fc086")

print("Message meta data:")
print(message_meta)
print("Message data:")
print(message)

File: bindings/nodejs/examples/07_get_message_data.js
// Copyright 2021-2022 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

// In this example we will send a message and get the data and metadata for it
async function run() {
    const { Client } = require('@iota/client');

    // client will connect to testnet by default
    const client = new Client({
        nodes: [
            {
                url: 'http://localhost:14265',
                auth: null,
                disabled: false,
            },
        ],
        localPow: true,
    });

    try {
        // Create message with no payload
        const message = await client.generateMessage();
        console.log('Message:', message, '\n');

        // Send message
        const messageId = await client.postMessage(message);

        const messageData = await client.getMessageData(messageId);
        const messageMetadata = await client.getMessageMetadata(messageId);

        console.log('Message data: ', messageData, '\n');
        console.log('Message metadata: ', messageMetadata, '\n');
    } catch (error) {
        console.log(error);
    }
}

run().then(() => process.exit());
